### Load CSV

In [15]:
csvfilename = "Ex_Files_Python_Data_Visualization/Ex_Files_Python_Data_Visualization/Exercise_Files/3_Pandas/data/car_financing.csv"

In [69]:
import pandas as pd
df = pd.read_csv(csvfilename)
df

Month  Starting Balance  Repayment  Interest Paid  Principal Paid  \
0        1          34689.96     687.23         202.93          484.30   
1        2          34205.66     687.23         200.10          487.13   
2        3          33718.53     687.23         197.25          489.98   
3        4          33228.55     687.23         194.38          492.85   
4        5          32735.70     687.23         191.50          495.73   
..     ...               ...        ...            ...             ...   
403     56           3951.11     796.01           9.54          786.47   
404     57           3164.64     796.01           7.64          788.37   
405     58           2376.27     796.01           5.74          790.27   
406     59           1586.00     796.01           3.83          792.18   
407     60            793.82     796.01           1.91          794.10   

     New Balance  term  interest_rate       car_type  
0       34205.66    60         0.0702  Toyota Sienna  
1       33718.53    60         0.0702  Toyota Sienna  
2       33228.55    60         0.0702  Toyota Sienna  
3       32735.70    60         0.0702  Toyota Sienna  
4       32239.97    60         0.0702  Toyota Sienna  
..           ...   ...            ...            ...  
403      3164.64    60         0.0290      VW Golf R  
404      2376.27    60         0.0290      VW Golf R  
405      1586.00    60         0.0290      VW Golf R  
406       793.82    60         0.0290      VW Golf R  
407        -0.28    60         0.0290      VW Golf R  

[408 rows x 9 columns]

### Load excel file

In [30]:
excelfilename = "Ex_Files_Python_Data_Visualization/Ex_Files_Python_Data_Visualization/Exercise_Files/3_Pandas/data/car_financing1.xls"

In [73]:
import pandas as pd
df = pd.read_excel(excelfilename)
df

Month  Starting Balance  Repayment  Interest Paid  Principal Paid  \
0        1          34689.96     687.23         202.93          484.30   
1        2          34205.66     687.23         200.10          487.13   
2        3          33718.53     687.23         197.25          489.98   
3        4          33228.55     687.23         194.38          492.85   
4        5          32735.70     687.23         191.50          495.73   
..     ...               ...        ...            ...             ...   
403     56           3951.11     796.01           9.54          786.47   
404     57           3164.64     796.01           7.64          788.37   
405     58           2376.27     796.01           5.74          790.27   
406     59           1586.00     796.01           3.83          792.18   
407     60            793.82     796.01           1.91          794.10   

     New Balance  term  interest_rate       car_type  
0       34205.66    60         0.0702  Toyota Sienna  
1       33718.53    60         0.0702  Toyota Sienna  
2       33228.55    60         0.0702  Toyota Sienna  
3       32735.70    60         0.0702  Toyota Sienna  
4       32239.97    60         0.0702  Toyota Sienna  
..           ...   ...            ...            ...  
403      3164.64    60         0.0290      VW Golf R  
404      2376.27    60         0.0290      VW Golf R  
405      1586.00    60         0.0290      VW Golf R  
406       793.82    60         0.0290      VW Golf R  
407        -0.28    60         0.0290      VW Golf R  

[408 rows x 9 columns]

In [33]:
df.head()

Month  Starting Balance  Repayment  Interest Paid  Principal Paid  \
0      1          34689.96     687.23         202.93          484.30   
1      2          34205.66     687.23         200.10          487.13   
2      3          33718.53     687.23         197.25          489.98   
3      4          33228.55     687.23         194.38          492.85   
4      5          32735.70     687.23         191.50          495.73   

   New Balance  term  interest_rate       car_type  
0     34205.66    60         0.0702  Toyota Sienna  
1     33718.53    60         0.0702  Toyota Sienna  
2     33228.55    60         0.0702  Toyota Sienna  
3     32735.70    60         0.0702  Toyota Sienna  
4     32239.97    60         0.0702  Toyota Sienna

In [34]:
df.tail()

Month  Starting Balance  Repayment  Interest Paid  Principal Paid  \
403     56           3951.11     796.01           9.54          786.47   
404     57           3164.64     796.01           7.64          788.37   
405     58           2376.27     796.01           5.74          790.27   
406     59           1586.00     796.01           3.83          792.18   
407     60            793.82     796.01           1.91          794.10   

     New Balance  term  interest_rate   car_type  
403      3164.64    60          0.029  VW Golf R  
404      2376.27    60          0.029  VW Golf R  
405      1586.00    60          0.029  VW Golf R  
406       793.82    60          0.029  VW Golf R  
407        -0.28    60          0.029  VW Golf R

In [40]:
df.dtypes

Month                 int64
Starting Balance    float64
Repayment           float64
Interest Paid       float64
Principal Paid      float64
New Balance         float64
term                  int64
interest_rate       float64
car_type             object
dtype: object

In [37]:
df.shape

(408, 9)

In [38]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 408 entries, 0 to 407
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Month             408 non-null    int64  
 1   Starting Balance  408 non-null    float64
 2   Repayment         408 non-null    float64
 3   Interest Paid     407 non-null    float64
 4   Principal Paid    408 non-null    float64
 5   New Balance       408 non-null    float64
 6   term              408 non-null    int64  
 7   interest_rate     408 non-null    float64
 8   car_type          408 non-null    object 
dtypes: float64(6), int64(2), object(1)
memory usage: 28.8+ KB


In [41]:
type(df)

pandas.core.frame.DataFrame

In [45]:
df[['Month','Starting Balance','Repayment']].head()

Month  Starting Balance  Repayment
0      1          34689.96     687.23
1      2          34205.66     687.23
2      3          33718.53     687.23
3      4          33228.55     687.23
4      5          32735.70     687.23

In [46]:
df[['Month','Starting Balance','Repayment']].tail()

Month  Starting Balance  Repayment
403     56           3951.11     796.01
404     57           3164.64     796.01
405     58           2376.27     796.01
406     59           1586.00     796.01
407     60            793.82     796.01

In [49]:
type(df[['Month','Starting Balance','Repayment']].head())

pandas.core.frame.DataFrame

### Slicing

In [53]:
df[['Month','Starting Balance','Repayment']][5:10]

Month  Starting Balance  Repayment
5      6          32239.97     687.23
6      7          31741.34     687.23
7      8          31239.79     687.23
8      9          30735.31     687.23
9     10          30227.88     687.23

### Filtering

### I want to see only car_type = Toyota Sienna and interest rate = 7.02%

In [55]:
df['car_type'].value_counts()

VW Golf R         144
Toyota Sienna     120
Toyota Carolla    111
Toyota Corolla     33
Name: car_type, dtype: int64

In [58]:
df['car_type']=='Toyota Sienna'

0       True
1       True
2       True
3       True
4       True
       ...  
403    False
404    False
405    False
406    False
407    False
Name: car_type, Length: 408, dtype: bool

In [59]:
car_filter=df['car_type']=='Toyota Sienna'

In [60]:
df['interest_rate'].value_counts()

0.0290    144
0.0390    144
0.0359     60
0.0702     60
Name: interest_rate, dtype: int64

In [62]:
df['interest_rate']==0.0702

0      True
1      True
2      True
3      True
4      True
       ... 
403    True
404    True
405    True
406    True
407    True
Name: interest_rate, Length: 408, dtype: bool

In [63]:
interest_filter=df['interest_rate']==0.0702

In [81]:
df[car_filter]

Month  Starting Balance  Repayment  Interest Paid  Principal Paid  \
0        1          34689.96     687.23         202.93          484.30   
1        2          34205.66     687.23         200.10          487.13   
2        3          33718.53     687.23         197.25          489.98   
3        4          33228.55     687.23         194.38          492.85   
4        5          32735.70     687.23         191.50          495.73   
..     ...               ...        ...            ...             ...   
115     56           3133.83     632.47           9.37          623.10   
116     57           2510.73     632.47           7.51          624.96   
117     58           1885.77     632.47           5.64          626.83   
118     59           1258.94     632.47           3.76          628.71   
119     60            630.23     632.47           1.88          630.59   

     New Balance  term  interest_rate       car_type  
0       34205.66    60         0.0702  Toyota Sienna  
1       33718.53    60         0.0702  Toyota Sienna  
2       33228.55    60         0.0702  Toyota Sienna  
3       32735.70    60         0.0702  Toyota Sienna  
4       32239.97    60         0.0702  Toyota Sienna  
..           ...   ...            ...            ...  
115      2510.73    60         0.0359  Toyota Sienna  
116      1885.77    60         0.0359  Toyota Sienna  
117      1258.94    60         0.0359  Toyota Sienna  
118       630.23    60         0.0359  Toyota Sienna  
119        -0.36    60         0.0359  Toyota Sienna  

[120 rows x 9 columns]

In [82]:
df[interest_filter]

Month  Starting Balance  Repayment  Interest Paid  Principal Paid  \
0        1          34689.96     687.23         202.93          484.30   
1        2          34205.66     687.23         200.10          487.13   
2        3          33718.53     687.23         197.25          489.98   
3        4          33228.55     687.23         194.38          492.85   
4        5          32735.70     687.23         191.50          495.73   
..     ...               ...        ...            ...             ...   
403     56           3951.11     796.01           9.54          786.47   
404     57           3164.64     796.01           7.64          788.37   
405     58           2376.27     796.01           5.74          790.27   
406     59           1586.00     796.01           3.83          792.18   
407     60            793.82     796.01           1.91          794.10   

     New Balance  term  interest_rate       car_type  
0       34205.66    60         0.0702  Toyota Sienna  
1       33718.53    60         0.0702  Toyota Sienna  
2       33228.55    60         0.0702  Toyota Sienna  
3       32735.70    60         0.0702  Toyota Sienna  
4       32239.97    60         0.0702  Toyota Sienna  
..           ...   ...            ...            ...  
403      3164.64    60         0.0290      VW Golf R  
404      2376.27    60         0.0290      VW Golf R  
405      1586.00    60         0.0290      VW Golf R  
406       793.82    60         0.0290      VW Golf R  
407        -0.28    60         0.0290      VW Golf R  

[408 rows x 9 columns]

In [83]:
df[car_filter & interest_filter]

Month  Starting Balance  Repayment  Interest Paid  Principal Paid  \
0        1          34689.96     687.23         202.93          484.30   
1        2          34205.66     687.23         200.10          487.13   
2        3          33718.53     687.23         197.25          489.98   
3        4          33228.55     687.23         194.38          492.85   
4        5          32735.70     687.23         191.50          495.73   
..     ...               ...        ...            ...             ...   
115     56           3133.83     632.47           9.37          623.10   
116     57           2510.73     632.47           7.51          624.96   
117     58           1885.77     632.47           5.64          626.83   
118     59           1258.94     632.47           3.76          628.71   
119     60            630.23     632.47           1.88          630.59   

     New Balance  term  interest_rate       car_type  
0       34205.66    60         0.0702  Toyota Sienna  
1       33718.53    60         0.0702  Toyota Sienna  
2       33228.55    60         0.0702  Toyota Sienna  
3       32735.70    60         0.0702  Toyota Sienna  
4       32239.97    60         0.0702  Toyota Sienna  
..           ...   ...            ...            ...  
115      2510.73    60         0.0359  Toyota Sienna  
116      1885.77    60         0.0359  Toyota Sienna  
117      1258.94    60         0.0359  Toyota Sienna  
118       630.23    60         0.0359  Toyota Sienna  
119        -0.36    60         0.0359  Toyota Sienna  

[120 rows x 9 columns]

In [84]:
df=df[car_filter & interest_filter]

In [85]:
df

Month  Starting Balance  Repayment  Interest Paid  Principal Paid  \
0        1          34689.96     687.23         202.93          484.30   
1        2          34205.66     687.23         200.10          487.13   
2        3          33718.53     687.23         197.25          489.98   
3        4          33228.55     687.23         194.38          492.85   
4        5          32735.70     687.23         191.50          495.73   
..     ...               ...        ...            ...             ...   
115     56           3133.83     632.47           9.37          623.10   
116     57           2510.73     632.47           7.51          624.96   
117     58           1885.77     632.47           5.64          626.83   
118     59           1258.94     632.47           3.76          628.71   
119     60            630.23     632.47           1.88          630.59   

     New Balance  term  interest_rate       car_type  
0       34205.66    60         0.0702  Toyota Sienna  
1       33718.53    60         0.0702  Toyota Sienna  
2       33228.55    60         0.0702  Toyota Sienna  
3       32735.70    60         0.0702  Toyota Sienna  
4       32239.97    60         0.0702  Toyota Sienna  
..           ...   ...            ...            ...  
115      2510.73    60         0.0359  Toyota Sienna  
116      1885.77    60         0.0359  Toyota Sienna  
117      1258.94    60         0.0359  Toyota Sienna  
118       630.23    60         0.0359  Toyota Sienna  
119        -0.36    60         0.0359  Toyota Sienna  

[120 rows x 9 columns]

In [88]:
df[['car_type','interest_rate']].value_counts()

car_type       interest_rate
Toyota Sienna  0.0702           60
               0.0359           60
dtype: int64